### PR TITLE
Allow the layout to control how windows are raised in groups

### PR DESCRIPTION
**Pull request by outfoxxed**
#### Describe your PR, what does it fix/add?
Previously windows could only be focused if they weren't hidden or were part of a group. This shifts the logic for identifying group windows and showing them to the layout allowing for alternate group implementations to function normally.

Needed to fix outfoxxed/hy3#28 ([#3be4f05](outfoxxed/hy3/commit/3be4f0556bc47f388c6cd128199001c6b6eea3ad))

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
No

#### Is it ready for merging, or does it need work?
Ready

### CODE REVIEW

File: src/layout/IHyprLayout.hpp [hunk 1]
```diff
@@ -155,9 +155,23 @@ class IHyprLayout {
     */
     virtual void replaceWindowDataWith(CWindow* from, CWindow* to) = 0;
 
+    /*
+        Determines if a window can be focused. If hidden this usually means the window is part of a group.
+		*/
```
Comment (vaxerski) [inline src/layout/IHyprLayout.hpp:160]: why is the indent here so weird? with the other too

File: src/layout/IHyprLayout.hpp [hunk 2]
```diff
@@ -155,9 +155,23 @@ class IHyprLayout {
...
+        Brings the window to the top of any groups and ensures it is not hidden.
+
+        If the window is unmapped following this call, the focus attempt will fail.
+	  */
+    virtual void bringWindowToTop(CWindow*);
+
     /*
         Called via the foreign toplevel activation protocol.
         Focuses a window, bringing it to the top of its group if applicable.
+
```
Comment (vaxerski) [inline src/layout/IHyprLayout.hpp:174]: ?

File: src/layout/IHyprLayout.hpp (hunk 1 above)
Comment (outfoxxed) [reply src/layout/IHyprLayout.hpp:160]: what do you mean?
image

File: src/layout/IHyprLayout.hpp (hunk 2 above)
Comment (outfoxxed) [reply src/layout/IHyprLayout.hpp:174]: I'll fix it ig

File: src/layout/IHyprLayout.hpp (hunk 1 above)
Comment (outfoxxed) [reply src/layout/IHyprLayout.hpp:160]: It appears clang-format dosen't remove tabs in comments, sorry
image

**Review by vaxerski — APPROVED** [review]
ok, thanks!

**Pull request merged**